findimports.py

```diff
@@ -746,17 +748,26 @@ def listModules(self):
         modules.sort()
         return [module for name, module in modules]
 
-    def packageGraph(self, packagelevel=None):
+    def packageGraph(self, packagelevel=None, externals_only=False):
         """Convert a module graph to a package graph."""
         packages = {}
-        for module in self.listModules():
-            package_name = self.packageOf(module.modname, packagelevel)
-            if package_name not in packages:
-                dirname = os.path.dirname(module.filename)
-                packages[package_name] = Module(package_name, dirname)
-            package = packages[package_name]
+        modules = self.listModules()
+        internal_names = {m.modname for m in modules}
+        for module in modules:
+            if externals_only:
+                package = Module(module.modname, module.filename)
+                packages[module.modname] = package
+            else:
+                package_name = self.packageOf(module.modname, packagelevel)
+                if package_name not in packages:
+                    dirname = os.path.dirname(module.filename)
+                    packages[package_name] = Module(package_name, dirname)
+                package = packages[package_name]
             for name in module.imports:
-                package_name = self.packageOf(name, packagelevel)
+                if externals_only and name in internal_names:
+                    package_name = name
+                else:
+                    package_name = self.packageOf(name, packagelevel)
```

Comment:
This is a bit gnarly.  I would like a couple of extra functions, maybe `isExternal()` and `maybePackageOf(modname, packagelevel, externals_only)` and then hopefully the old look may suffice?

Reply:
Agree it's not ideal from design perspective, but the logic is working correctly there. Maybe leaving some notes for the next time some bigger work is done would suffice?

Reply:
That appeared to be more straight forward than I initially thought. done.